src/databricks/labs/dqx/datacontract/contract_rules_generator.py

```diff
@@ -0,0 +1,691 @@
+        if self.llm_engine is None:
+            return []
+
+        # Build context for LLM
+        context_info = f"Field: {column_path}"
+        if field.type:
+            context_info += f", Type: {field.type}"
+        if field.description:
+            context_info += f", Description: {field.description}"
+
+        user_input = f"{context_info}\nExpectation: {text_expectation}"
+
+        # Generate rules using LLM
+        logger.info(f"Processing text rule for {column_path}: {text_expectation}")
+        prediction = self.llm_engine.get_business_rules_with_llm(user_input=user_input, schema_info=schema_info)
+
+        return json.loads(prediction.quality_rules)
+
+    def _add_text_rule_metadata(
+        self, rule: dict, column_path: str, text_expectation: str, contract_metadata: dict
+    ) -> dict:
+        """Add contract metadata to a text-generated rule."""
+        if 'user_metadata' not in rule:
+            rule['user_metadata'] = {}
+        rule['user_metadata'].update(
+            {
+                **contract_metadata,
+                "field": column_path,
+                "rule_type": "text_llm",
+                "text_expectation": text_expectation,
+            }
+        )
+        return rule
+
+    def _process_explicit_rules_for_model(
+        self, model: "Model", model_name: str, spec: "DataContractSpecification", _default_criticality: str
+    ) -> list[dict]:
+        """Process explicit DQX format rules from a model."""
+        contract_metadata = {
+            "contract_id": spec.id or "unknown",
+            "contract_version": self._get_contract_version(spec),
+            "model": model_name,
+        }
+
+        rules = []
+        for field_name, field in (model.fields or {}).items():
+            rules.extend(self._extract_explicit_rules_for_field(field, field_name, contract_metadata))
+
+        return rules
+
+    def _extract_explicit_rules_for_field(
+        self, field: "Field", field_name: str, contract_metadata: dict, parent_path: str = ""
+    ) -> list[dict]:
+        """Extract explicit DQX rules from a field's quality checks."""
+        column_path = f"{parent_path}.{field_name}" if parent_path else field_name
+
+        # Recurse for nested fields
+        if field.fields:
+            rules = []
+            for nested_name, nested_field in field.fields.items():
+                rules.extend(
+                    self._extract_explicit_rules_for_field(nested_field, nested_name, contract_metadata, column_path)
+                )
+            return rules
+
+        # Extract explicit DQX rules from quality checks
+        if not field.quality:
+            return []
+
+        rules = []
+        for quality in field.quality:
+            if self._is_dqx_quality(quality):
+                dqx_rule = self._extract_dqx_rule_from_quality(quality, column_path, contract_metadata)
+                if dqx_rule:
+                    rules.append(dqx_rule)
```

Comment:
it would be good to perform validation before adding the rule using `DQEngine.validate_checks(dqx_rule)`